test/test_functional.py

```diff
@@ -63,7 +63,7 @@ def hello():
 
     pre_size = len(hello.__dict__)
     keys = set(hello.__dict__.keys())
-    data_hello = data(1, 2)(hello)
+    data_hello = data(1, 2)(data(3)(hello))
```

Comment:
I'd prefer adding a new test case for the new feature, rather than overloading an existing test case.